crates/biome_js_analyze/src/lint/correctness/no_unused_imports.rs

```diff
@@ -23,6 +32,21 @@ declare_rule! {
     /// the unused imports will also be removed. So that comment directives
     /// like `@ts-expect-error` won't be transferred to a wrong place.
     ///
+    /// ## Options
+    ///
+    /// The rule provides a single option `ignoreReact`.
+    /// When this option is set to `true`, imports named `React` from the package `react` are ignored.
+    /// `ignoreReact` is disabled by default.
+    ///
+    /// ```json
+    /// {
+    ///     "//": "...",
+    ///     "options": {
+    ///         "ignoreReact": true
+    ///     }
+    /// }
+    /// ```
```

Comment:
Maybe add a little explainer when this is necessary? Maybe:

> This should only be necessary if you cannot upgrade to a React version that supports the new JSX runtime.

Reply:
Good suggestion! Added :)